app/routes/calendar/$eventId.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { useParams } from '@remix-run/react';
+import EventModal from '~/components/common/eventModal';
+import { useStore } from '~/store/useStore';
+import { CalEvent } from '~/utils/interfaces';
+
+const EventDetails = () => {
+  const { events: eventsList } = useStore((state) => state);
+  const [calendarEvent, setCalendarEvent] = React.useState<any>({
+    title: '',
+    start: undefined,
+    end: undefined,
+    location: '',
+    description: '',
+    attendees: [],
+  });
+
+  const params = useParams();
+
+  React.useEffect(() => {
+    if (params.eventId !== 'new') {
+      const calEvent = eventsList.find(
+        (event: CalEvent) => event.id === parseInt(params.eventId as string, 10),
+      );
```

Comment:
handle the case where it does not find an event

Reply:
will try to handle `calendar/${unknownId}`